src/core/ext/filters/client_channel/client_channel.c

```diff
@@ -236,14 +236,19 @@ static void set_channel_connectivity_state_locked(grpc_exec_ctx *exec_ctx,
                                                   grpc_connectivity_state state,
                                                   grpc_error *error,
                                                   const char *reason) {
-  if ((state == GRPC_CHANNEL_TRANSIENT_FAILURE ||
-       state == GRPC_CHANNEL_SHUTDOWN) &&
-      chand->lb_policy != NULL) {
-    /* cancel picks with wait_for_ready=false */
-    grpc_lb_policy_cancel_picks_locked(
-        exec_ctx, chand->lb_policy,
-        /* mask= */ GRPC_INITIAL_METADATA_WAIT_FOR_READY,
-        /* check= */ 0, GRPC_ERROR_REF(error));
+  if (chand->lb_policy != NULL) {
+    if (state == GRPC_CHANNEL_TRANSIENT_FAILURE) {
```

Comment:
Let's put a TODO here about improving how we handle failure cases.  In particular:

1. At the moment, I suspect it's not possible for any LB policy to actually return TRANSIENT_FAILURE.  We should probably fix that, which I think will require changing round_robin to be able to proactively unref and recreate subchannels for failed connections.

2. If we are replacing the LB policy, we should ideally re-submit any pending picks to the new LB policy instead of failing them directly.